arbos/tx_processor.go

```diff
@@ -79,42 +79,50 @@ func (p *TxProcessor) StartTxHook() (endTxNow bool, gasUsed uint64, err error, r
 		return false, 0, nil, nil
 	}
 
-	evm := p.evm
-
+	var tracingInfo *util.TracingInfo
+	from := p.msg.From()
 	tipe := underlyingTx.Type()
 	p.TopTxType = &tipe
+	evm := p.evm
 
-	startTracer := func() func() {
+	startTracer := func(calling common.Address) func() {
 		if !evm.Config.Debug {
 			return func() {}
 		}
 		evm.IncrementDepth() // fake a call
 		tracer := evm.Config.Tracer
 		start := time.Now()
-		tracer.CaptureStart(evm, p.msg.From(), *p.msg.To(), false, p.msg.Data(), p.msg.Gas(), p.msg.Value())
+		tracer.CaptureStart(evm, from, *p.msg.To(), false, p.msg.Data(), p.msg.Gas(), p.msg.Value())
+
+		tracingInfo = util.NewTracingInfo(evm, from, calling, util.TracingDuringEVM)
```

Comment:
Is there a reason we're not using `*p.msg.To()` instead of `calling`?

Reply:
This was originally meant to let you fake it for things like the call to `redeem()` but since `MockCall` lets you specify your own `from` and `to` you're right, it's better as `*p.msg.To()`. I've changed it